src/services/dataFetcher.ts

```
@@ -159,219 +95,54 @@ export class DataFetcher {
         reason: errorMessage,
         details: logDetails,
       });
-
-      // Update lastDataRequest even on general error to prevent infinite retry loops
-      DataFetcher.setTileState<T>(storageKey, null, false);
-
+      // On error, update only lastDataRequest and lastDataRequestSuccessful, do not update data
+      const cached = storageManager.getTileState<TTileData>(storageKey);
+      const prevData =
+        cached && cached.data ? cached.data : (mapper.createDefault() as unknown as TTileData);
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)